gobblin-iceberg/src/main/java/org/apache/gobblin/iceberg/writer/IcebergMetadataWriter.java

```diff
@@ -684,7 +687,8 @@ protected void addFiles(GobblinMetadataChangeEvent gmce, Map<String, Collection<
         Collection<HiveSpec> hiveSpecs = newSpecsMap.get(new Path(file.getFilePath()).getParent().toString());
         StructLike partition = getIcebergPartitionVal(hiveSpecs, file.getFilePath(), partitionSpec);
 
-        if(tableMetadata.completenessEnabled && gmce.getOperationType() == OperationType.add_files) {
+        if(this.newPartitionTableWhitelistBlacklist.acceptTable(tableMetadata.dbName, tableMetadata.tableName)
+            && gmce.getOperationType() == OperationType.add_files) {
           tableMetadata.prevCompletenessWatermark = Long.parseLong(table.properties().getOrDefault(COMPLETION_WATERMARK_KEY,
```

Comment:
What will happen if it's not add_files operation? i.e. what will happen for compaction?

Reply:
we will only white list kafka_streaming_hourly tables and re_write files will never occur for them.

Reply:
Do we want to make this more general? we can just add one place holder here and address in another PR